src/components/NewMovie/NewMovie.tsx

```diff
@@ -1,45 +1,97 @@
 import { useState } from 'react';
 import { TextField } from '../TextField';
+import { Movie } from '../../types/Movie';
+import { validationPattern } from '../../utils/urlValidationPattern';
 
-export const NewMovie = () => {
+type Props = {
+  onAdd: (movie: Movie) => void;
+};
+
+const initialMovie = {
+  title: '',
+  description: '',
+  imgUrl: '',
+  imdbUrl: '',
+  imdbId: '',
+};
+
+export const NewMovie: React.FC<Props> = ({ onAdd }) => {
   // Increase the count after successful form submission
   // to reset touched status of all the `Field`s
-  const [count] = useState(0);
+  const [count, setCount] = useState(0);
+  const [movie, setMovie] = useState<Movie>(initialMovie);
+  const {
+    title,
+    description,
+    imgUrl,
+    imdbUrl,
+    imdbId,
+  } = movie;
+  const isEmpty = !title.trim()
+  || !imdbId.trim()
+  || !imdbUrl.trim()
+  || !imgUrl.trim();
+  const isPassingValidation = validationPattern.test(imdbUrl)
+    && validationPattern.test(imgUrl);
+
+  const handleChangeFieldValue = (
+    fieldName: keyof Movie,
+    fieldValue: string,
+  ) => {
+    setMovie((prev) => ({ ...prev, [fieldName]: fieldValue }));
+  };
+
+  const handleSubmit = (event: React.FormEvent<HTMLFormElement>) => {
+    event.preventDefault();
+    onAdd(movie);
+
+    setCount((currentCount) => currentCount + 1);
+    setMovie(initialMovie);
+  };
 
   return (
-    <form className="NewMovie" key={count}>
+    <form className="NewMovie" key={count} onSubmit={handleSubmit}>
       <h2 className="title">Add a movie</h2>
 
       <TextField
         name="title"
         label="Title"
-        value=""
-        onChange={() => {}}
+        value={title}
+        onChange={(newTitle) => handleChangeFieldValue('title', newTitle)}
```

Comment:
If make changes as I wrote above, you can simplify this

```suggestion
        onChange={handleChangeFieldValue)}
```

The same in other places